packages/react-devtools-shared/src/frontend/utils/withPermissionsCheck.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import {PermissionNotGrantedError} from 'react-devtools-shared/src/errors/PermissionNotGrantedError';
+
+type SupportedPermission = 'clipboardWrite';
+type Permissions = Array<SupportedPermission>;
+type PermissionsOptions = {permissions: Permissions};
+
+// browser.permissions is not available for DevTools pages in Firefox
+// https://bugzilla.mozilla.org/show_bug.cgi?id=1796933
+// We are going to assume that requested permissions are not optional.
+export function withPermissionsCheck<T: (...$ReadOnlyArray<empty>) => mixed>(
+  options: PermissionsOptions,
+  callback: T,
+): T | (() => Promise<ReturnType<T>>) {
+  if (!__IS_CHROME__ && !__IS_EDGE__) {
+    return callback;
+  } else {
+    return async () => {
+      const alreadyContains = await chrome.permissions.contains(options);
```

Comment:
Do we need this check? I'd expect that `chrome.permissions.request` does not prompt if you already granted permissions. At least that's how I interpret "displaying a prompt to the user **if necessary**." (https://developer.chrome.com/docs/extensions/reference/api/permissions#method-request)

Reply:
Yes, I believe when I was just requesting the permission, it was showing me the dialog window each time. I will double-check later today and follow-up on this with you.

Reply:
Checked that again, the request dialog is not shown for permissions that already granted, thanks for pointing out! Removed the check.